# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/PathHelpers.h
# Commit: 6ea4f3bba40f
# Full Hash: 6ea4f3bba40f2fdf5c4893e0eb76fee7d4e05094
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2025-07-05 09:22:06
# Regressor Bug: 1912897
# File Overlap Count: 1
# Description:
#   Bug 1912897 - Support blurring and shadows for Accelerated Canvas2D. r=aosmond
#   
#   This adds support via FilterNodeWebgl for accelerating specific filters in AC2D.
#   In particular, it adds support for accelerated blurring using a separable blur
#   filter. This is then used to implement the Gaussian Blur filter as well as shadows.
# ==============================================================================

diff -r c6d00b59234d -r 6ea4f3bba40f gfx/2d/PathHelpers.h
--- a/gfx/2d/PathHelpers.h	Sat Jul 05 00:10:54 2025 +0000
+++ b/gfx/2d/PathHelpers.h	Sat Jul 05 04:41:59 2025 +0000
@@ -153,8 +153,13 @@
   }
 }
 
+inline already_AddRefed<Path> MakeEmptyPath(const DrawTarget& aDrawTarget) {
+  RefPtr<PathBuilder> builder = aDrawTarget.CreatePathBuilder();
+  return builder->Finish();
+}
+
 /**
- * Appends a path represending a rectangle to the path being built by
+ * Appends a path representing a rectangle to the path being built by
  * aPathBuilder.
  *
  * aRect           The rectangle to append.
@@ -175,7 +180,7 @@
 }
 
 /**
- * Appends a path represending a rounded rectangle to the path being built by
+ * Appends a path representing a rounded rectangle to the path being built by
  * aPathBuilder.
  *
  * aRect           The rectangle to append.
@@ -199,7 +204,7 @@
 }
 
 /**
- * Appends a path represending an ellipse to the path being built by
+ * Appends a path representing an ellipse to the path being built by
  * aPathBuilder.
  *
  * The ellipse extends aDimensions.width / 2.0 in the horizontal direction